packages/dashboard/src/customization/propertiesSections/settings/section.tsx

```diff
@@ -22,18 +22,34 @@ export const SettingsSection = ({
   significantDigits: number | undefined;
   updateSignificantDigits: (newValue: number | undefined) => void;
 }) => {
-  const onSignificantDigitsChange: NonCancelableEventHandler<
-    InputProps.ChangeDetail
-  > = (event) => {
-    updateSignificantDigits(
-      isNumeric(event.detail.value) ? numberFromDetail(event) : undefined
-    );
+  const { control, clearErrors } = useForm<{
+    decimalPlaces: string;
+  }>({ mode: 'onChange' });
+
+  const [decimalPlacesValue, setDecimalPlacesValue] = useState<
+    number | undefined
+  >(significantDigits);
+
+  const selectedWidgets = useSelectedWidgets();
+  const selectedWidgetId = selectedWidgets[0]?.id;
+
+  useEffect(() => {
+    setDecimalPlacesValue(significantDigits);
+    clearErrors();
```

Comment:
`react-hook-form` should remove the error automatically when none of the validation rules are broken. Why do we need this effect to manually clear the error?

Reply:
I'm using `mode: 'onChange',` and react-hook-form is not clearing the errors when a different widget is selected.
So had to do it manually if selectedWidgetId is changed

This is the issue I'm facing if I don't manually clear the errors
[decimal-places-clear-error-issue.webm](https://github.com/awslabs/iot-app-kit/assets/154328164/c41abdbe-4224-435c-9263-72f34258ad41)